libraries/chain/include/graphene/chain/protocol/transaction.hpp

```diff
@@ -123,6 +123,13 @@ namespace graphene { namespace chain {
       signed_transaction( const transaction& trx = transaction() )
          : transaction(trx){}
 
+      /** Extract public keys from signatures when initializing with another signed transaction and a chain ID */
+      signed_transaction( const signed_transaction& trx, const chain_id_type& chain_id )
+         : signed_transaction(trx)
+      {
+         signees = _get_signature_keys( chain_id );
+      }
+
```

Comment:
Please add move constructor and move assignment operator, for moving signees and signatures efficiently.

Reply:
Won't move constructor and move assignment operator be implicitly added by compiler? I think this constructor is not a copy constructor, so won't prevent compilers from doing so.

Reply:
Oops, you're right.